### DISCUSSION TITLE
Coupling FSI module with XFEM - filling crack with a fluid

### DISCUSSION THREAD

**Discussion by Hyunkel07** [idaholab/moose#22890] [Q&A Modules: General] [answered]
Hello Everyone,

I am Rémi a researcher in Volcanology. I am trying to model the evolution of magmatic intrusions (fluid) in a solid rock. The magma is injected in the solid rock through a crack that should propagate when failure condition of the host-rock are fulfilled. I have been looking around to find a code that can solve propagation of pressurized magma in a continuum domain (the solid rock). It seems that coupling the FSI and XFEM modules in MOOSE could work.
But I would like to ask you directly. Can cracks formed with the XFEM module be filled with a material ? (in my case a non newtonian or newtonian fluid that represents the magma)

best regards ! :)

Rémi

**Comment by hugary1995** [comment, marked as answer]
Hi,

> Can cracks form with the XFEM module be filled with a material ?

Yes and no. No in the sense that no additional geometry can be created on the fly to represent the fluid. Yes, if you can represent the state of the fluid using some state variables which live on the lower dimension XFEM interface.

**Reply by Hyunkel07** [reply]
Hi,

Thank you for this quick answer.

>  Yes, if you can represent the state of the fluid using some state variables which live on the lower dimension XFEM interface.

Ok, so I reckon that the fluid won't be properly modelled in the crack, by that I mean that the velocity field and pressure state cannot be solved in the lower dimension XFEM interface... or ?

Rémi

**Reply by GiudGiud** [reply]
Are you planning to solve for the velocity and pressure field in 3D? Or would a lower dimensional domain inside the crack be OK ?

I ll say that this has not been done before. So it will involve development. I dont see a roadblock in MOOSE for doing that.

The FSI module was not developed for a moving or expanding domain. It can handle small displacements of the nodes at each time step, it does not have the capability to add a new element for example, I feel that there needs to be some treatment in the equation to be able to add elements while conserving at least mass.

**Reply by hugary1995** [reply]
yeah, this is definitely doable using the existing infrastructure, it just requires you to do development (knowing some c++ and willing to write code). I know @jiangwen84 has done something similar.

**Reply by Hyunkel07** [reply]
> Are you planning to solve for the velocity and pressure field in 3D? Or would a lower dimensional domain inside the crack be OK ?

The plan is to start in 2D (something similar than the attached picture) but if the results are convincing I would like to extend the study to 3D examples.

This is a laccolith model, where the liquid magma (embedded in a solid host-rock) is injected in a pipe from the bottom and circulate in a chamber that inflates as the pressure increases.

![model_laccolith](https://user-images.githubusercontent.com/58592723/206142180-8010a952-fa34-40c9-b177-dc573443c23a.PNG)

My plan is to do something similar than above but to have the pressurized magma propagating through the host rock when failure condition are fulfilled. In this example, the inflation of the magma cavity (the ellipse), induce stress concentration in the host-rock at the tip of the ellipse. In nature, the host-rock would break where stress concentration occurs, creating space for the magma to propagate.

> yeah, this is definitely doable using the existing infrastructure, it just requires you to do development (knowing some c++ and willing to write code)

I will try to look at code development and assess if it is doable for me :)

Best regards,

Rémi

**Comment by friedmud** [comment]
To me - this doesn't seem like a case for XFEM... it seems like a continuum model would be better.  Something like PhaseField... or using the LevelSet module.
